### PR TITLE
Update and fix demo

### PR DESCRIPTION
**Pull request by luizstacio**
### Summary

- Update Contract dependencies, lock `sway-lib-std` and `sway-lib-core`
- Fix amounts to work properly
- Move from color to `asset_id`
- Add `docker` images, to enable local development with local node
- Improve Provider configuration using `.env` file

### CODE REVIEW

File: client/src/components/Assets.tsx [hunk 1]
```diff
@@ -70,9 +72,9 @@ export const Assets = () => {
         </div>
 
         {transformCoinsToAssets(coins).map((coinData) => (
-          <div className={style.transferPropContainer} key={coinData.color}>
+          <div className={style.transferPropContainer} key={coinData.assetId}>
             <div className={style.transferPropInput}>
-              <span>{coinData.amount}</span>
+              <span>{formatEther(coinData.amount)}</span>
```
Comment (adlerjohn) [inline client/src/components/Assets.tsx:77]: Is this actually okay? We won't be using 18 decimals, we'll be using 9 at most.

Comment (luizstacio) [reply client/src/components/Assets.tsx:77]: Do we have a spec for this? If the max is 9, I can create a very similar utility.

Comment (adlerjohn) [reply client/src/components/Assets.tsx:77]: There's no spec for decimals on fungible tokens yet. But just based on the fact that we use `u64` there's no way to handle 18 decimals.

Comment (adlerjohn) [reply client/src/components/Assets.tsx:77]: Are we waiting for the above issue to be merged?

Comment (luizstacio) [reply client/src/components/Assets.tsx]: Just push a temp fix for it

**Review by adlerjohn — APPROVED** [review]
Haven't tested it, but I'll trust you

**Pull request merged**